pkg/apis/devops/v1alpha2/register.go

```diff
@@ -35,21 +37,71 @@ var (
 )
 
 func addWebService(c *restful.Container) error {
+
 	webservice := runtime.NewWebService(GroupVersion)
 
-	tags := []string{"devops"}
+	tags := []string{"DevOps"}
+
+	webservice.Route(webservice.GET("/devops/{devops}").
+		To(devopsapi.GetDevOpsProjectHandler).
+		Doc("get devops project").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Param(webservice.PathParameter("devops", "projectId")).
+		Writes(&devops.DevOpsProject{}))
+
+	webservice.Route(webservice.PATCH("/devops/{devops}").
+		To(devopsapi.UpdateProjectHandler).
+		Doc("get devops project").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Param(webservice.PathParameter("devops", "projectId")).
+		Writes(&devops.DevOpsProject{}))
+
+	webservice.Route(webservice.GET("/devops/{devops}/defaultroles").
+		To(devopsapi.GetDevOpsProjectDefaultRoles).
+		Doc("get devops project defaultroles").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Writes(&devops.DefaultRoles))
+
+	webservice.Route(webservice.GET("/devops/{devops}/members").
+		To(devopsapi.GetDevOpsProjectMembersHandler).
+		Doc("get devops project members").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Writes(&[]*devops.DevOpsProjectMembership{}))
+
+	webservice.Route(webservice.GET("/devops/{devops}/members/{members}").
```

Comment:
normally, parameter name use singular, like `member`

Reply:
In k8s, the plural of the resource is always used in the path.